source/MRViewer/MRObjectTransformWidget.h

```diff
@@ -53,9 +53,22 @@ class MRVIEWER_CLASS ObjectTransformWidget : public MultiListener<MouseDownListe
     void setPickThrough( bool on ) { pickThrough_ = on; }
     bool getPickThrough() const { return pickThrough_; }
 
+    enum AxisTransformMode
+    {
+        Translation,
+        Scaling,
+    };
+    AxisTransformMode getAxisTransformMode() const { return axisTransformMode_; };
+    void setAxisTransformMode( AxisTransformMode mode ) { axisTransformMode_ = mode; };
+
     // Returns root object of widget
     std::shared_ptr<Object> getRootObject() const { return controlsRoot_; }
 
+    // Changes controls xf (controls will affect object in basis of new xf)
+    // note that rotation is applied around 0 coordinate in world space, so use xfAround to process rotation around user defined center
+    void setControlsXf( const AffineXf3f& xf );
```

Comment:
public methods with hidden implementation should have `MRVIEWER_API`

Reply:
Fixed.